cylc/flow/workflow_files.py

```diff
@@ -620,8 +621,29 @@ def get_workflow_srv_dir(id_):
     return os.path.join(run_d, WorkflowFiles.Service.DIRNAME)
 
 
+def refresh_nfs_cache(path: Path):
+    """Refresh NFS cache for dirs between ~/cylc-run and <path> inclusive.
+
+    On NFS filesystems, the non-existence of files/directories may become
+    cashed. To work around this, we can list the contents of these directories
+    which refreshes the NFS cache.
+
+    See: https://github.com/cylc/cylc-flow/issues/6506
+
+    Arguments:
+        path: The directory to refresh.
+
+    Raises:
+        FileNotFoundError: If any of the directories between ~/cylc-run and
+        this directory (inclsive) are not present.
+
+    """
+    cylc_run_dir = get_cylc_run_dir()
+    for subdir in reversed(path.relative_to(cylc_run_dir).parents):
+        list((cylc_run_dir / subdir).iterdir())
```

Comment:
Is is necessary to list all adjacent files along the hierarchy down to the one we care about?

Just thinking about a big job log hierarchy.

Maybe use this to avoid loading into memory
```suggestion
    for subdir in reversed(path.relative_to(cylc_run_dir).parents):
        deque((cylc_run_dir / subdir).iterdir(), maxlen=0)
```

https://docs.python.org/3/library/itertools.html#itertools-recipes